### PR TITLE
Upgrade storybook to v8

### PR DESCRIPTION
**Pull request by yivlad**

### CODE REVIEW

File: packages/app/.storybook/main.ts [hunk 1]
```diff
@@ -6,31 +6,39 @@ import dotenv from 'dotenv'
...
     return env.parsed!
   },
+
   viteFinal: (config) =>
     mergeConfig(config, {
       plugins: [svgr()],
     }),
+
+  typescript: {
+    reactDocgen: 'react-docgen-typescript',
```
Comment (krzkaczor) [inline packages/app/.storybook/main.ts:40]: we need that? if it's just for docs we can skip it.

File: packages/app/src/features/actions/components/actions-grid/stories/AllHandlerStates.stories.ts [hunk 2]
```diff
@@ -214,8 +265,14 @@ export const ExchangeExtended: Story = {
     variant: 'extended',
   },
 }
-export const ExchangeExtendedMobile = { name: 'Exchange (Extended, Mobile)', ...getMobileStory(ExchangeExtended) }
-export const ExchangeExtendedTablet = { name: 'Exchange (Extended, Tablet)', ...getTabletStory(ExchangeExtended) }
+export const ExchangeExtendedMobile = {
```
Comment (krzkaczor) [inline packages/app/src/features/actions/components/actions-grid/stories/AllHandlerStates.stories.ts:268]: are these only formatting changes?

Comment (oskarvu) [reply packages/app/src/features/actions/components/actions-grid/stories/AllHandlerStates.stories.ts:268]: Did you trigger format manually (by adding new line)? When I run `biome format` it doesn't list these files as candidates for formatting.

File: packages/app/package.json [hunk 3]
```diff
@@ -125,8 +124,9 @@
     "postcss": "^8.4.26",
     "prop-types": "^15.8.1",
     "serve": "^14.2.1",
-    "storybook": "^7.5.1",
-    "storybook-addon-pseudo-states": "^2.1.2",
+    "storybook": "^8.1.4",
+    "storybook-addon-pseudo-states": "^3.1.1",
+    "storybook-addon-remix-react-router": "^3.0.0",
```
Comment (oskarvu) [inline packages/app/package.json:129]: Why `remix router` instead of previous `react router v6`?

**Review by oskarvu — APPROVED** [review]

File: packages/app/package.json (hunk 3 above)
Comment (yivlad) [reply packages/app/package.json:129]: The package was renamed, it's just newer version of it

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**